src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1562,7 +1562,7 @@ internal SynthesizedAttributeData SynthesizeRequiresLocationAttribute(ParameterS
             if ((object)Compilation.SourceModule != symbol.ContainingModule)
             {
                 // For symbols that are not defined in the same compilation (like NoPia), don't synthesize this attribute.
-                Debug.Assert(false); // PROTOTYPE: Test this code path.
+                Debug.Assert(false, $"Unexpected embedding of a parameter symbol '{symbol.ToDisplayString()}'");
```

Comment:
Don't think this path is reachable - only types are embedded in NoPIA scenarios, not parameters.

Reply:
>  only types are embedded in NoPIA scenarios, not parameters.

That is not accurate. Interface methods are embedded and their parameters as well. We probably have a test that hits this code path in ```SynthesizeIsReadOnlyAttribute```

Reply:
You're right, thanks, adding a test.